src/components/create-study-form.js

```diff
@@ -48,6 +54,8 @@ const useStyles = makeStyles(() => ({
     },
 }));
 
+let typingTimer;
```

Comment:
Using global variables is discouraged. For example, if we have 2 instances of the component, this variable is shared among them.